src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
Any other arguments we might need so we don't have to add yet another RPC later?

One that I can think of: do/don't change the principal's kvno.